guides/release/components/actions-and-events.md

```diff
@@ -0,0 +1,609 @@
+in a messaging application. Therefore within the component's JavaScript file, we
+will use a property `confirmValue` to represent that argument and pass it to
+`onConfirm` as shown here:
+
+```javascript {data-filename=app/components/button-with-confirmation.js}
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+
+export default class ButtonWithConfirmation extends Component {
+  @action
+  submitConfirm() {
+    if (this.args.onConfirm) {
+      // call `onConfirm` with a second argument
+      await this.args.onConfirm(this.confirmValue);
+    }
+
+    this.showConfirmation = false;
+  }
+
+  //...
+}
+```
+
+In order for `confirmValue` to take on the value of the message text, we'll bind
+the property to the value of a user input field that will appear when the button
+is clicked. To accomplish this, we'll first modify the component so that it can
+be used in block form and we will [yield](../yields/) `confirmValue` to the
+block within the `"confirmDialog"` element:
+
+```handlebars {data-filename=app/templates/components/button-with-confirmation.hbs}
+<button type="button" onclick={{this.launchConfirmDialog}}>{{this.text}}</button>
+
+{{#if this.confirmShown}}
+  <div class="confirm-dialog">
+    {{yield this.confirmValue}}
+    <button class="confirm-submit" type="button" onclick={{this.submitConfirm}}>OK</button>
+    <button class="confirm-cancel" type="button" onclick={{this.cancelConfirm}}>Cancel</button>
+  </div>
+{{/if}}
+```
+
+With this modification, we can now use the component in `SendMessage` to wrap a
+text input element whose `value` attribute is set to `confirmValue`:
+
+```handlebars {data-filename=app/templates/components/send-message.hbs}
+<ButtonWithConfirmation
+  @text="Click to send your message."
+  @onConfirm={{action this.sendMessage "info"}}
+as |confirmValue|>
+  {{input value=confirmValue}}
+</ButtonWithConfirmation>
+```
+
+When the user enters their message into the input field, the message text will
+now be available to the component as `confirmValue`. Then, once they click the
+"OK" button, the `submitConfirm` action will be triggered, calling `onConfirm`
+with the provided `confirmValue`, thus invoking the `sendMessage` action in
+`SendMessage` with both the `messageType` and `messageText` arguments.
+
+## Invoking Actions Directly on Component Collaborators
+
+Actions can be invoked on objects other than the component directly from the
+template. For example, in our `SendMessage` component we might include a service
+that processes the `sendMessage` logic.
+
+```javascript {data-filename=app/components/send-message.js}
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+
+export default class SendMessage extends Component {
+  @service
+  messaging;
+
+  // component implementation
+}
+```
+
+We can tell the action to invoke the `sendMessage` action directly on the
+messaging service with the `target` attribute.
```

Comment:
I don't see a `target` here

Reply:
This should be removed, yeah. The action is now bound directly on the service, so `target` is not needed.